billing-service/src/test/java/com/petclinic/billing/presentationlayer/BillResourceTest.java

```diff
@@ -0,0 +1,150 @@
+
+        when(service.CreateBill(any())).thenThrow(new NotFoundException("Bill not found"));
+
+        mvc.perform(post("/bills", 65)
+                .content(objectMapper.writeValueAsString(newDTO))
+                .contentType(MediaType.APPLICATION_JSON)
+                .characterEncoding("utf-8")
+                .accept(MediaType.APPLICATION_JSON))
+				.andExpect(status().isNotFound())
+                .andExpect(result -> assertTrue(result.getResolvedException() instanceof NotFoundException))
+                .andExpect(result -> assertEquals("Bill not found", result.getResolvedException().getMessage()));
+    }
+
+    @Test
+    void createBillInvalidInputNegativeNumber() throws Exception {
+        BillDTO newDTO = new BillDTO(-1, 1, new Date(), "type1", 1.0);
+
+        when(service.CreateBill(any())).thenThrow(new InvalidInputException("That bill id does not exist"));
+
+        mvc.perform(post("/bills", 1)
+                        .content(objectMapper.writeValueAsString(newDTO))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .characterEncoding("utf-8")
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isUnprocessableEntity())
+                .andExpect(result -> assertTrue(result.getResolvedException() instanceof InvalidInputException))
+                .andExpect(result -> assertEquals("That bill id does not exist", result.getResolvedException().getMessage()));
+    }
+
+    @Test
+    void createBillInvalidInputZero() throws Exception {
+        BillDTO newDTO = new BillDTO(0, 1, new Date(), "type1", 1.0);
+
+        when(service.CreateBill(any())).thenThrow(new InvalidInputException("That bill id does not exist"));
+
+        mvc.perform(post("/bills", 1)
+                        .content(objectMapper.writeValueAsString(newDTO))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .characterEncoding("utf-8")
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isUnprocessableEntity())
+                .andExpect(result -> assertTrue(result.getResolvedException() instanceof InvalidInputException))
+                .andExpect(result -> assertEquals("That bill id does not exist", result.getResolvedException().getMessage()));
+    }
+
+    @Test
+    void findBill() {
+
+    }
+
+    @Test
+    void deleteBill() {
+
+    }
+
+    @Test
+    void testEmptyHttpErrorInfo() {
+        HttpErrorInfo httpErrorInfo = new HttpErrorInfo();
+
+        assertEquals(httpErrorInfo.getHttpStatus(), null);
+        assertEquals(httpErrorInfo.getPath(), null);
+        assertEquals(httpErrorInfo.getTimestamp(), null);
+        assertEquals(httpErrorInfo.getMessage(), null);
+    }
+
+    @Test
+    void testHttpErrorInfoConstructor() {
+        HttpErrorInfo httpErrorInfo = new HttpErrorInfo("timestamp1", HttpStatus.NOT_FOUND, "Bill not found");
+    }
```

Comment:
I don't think this test tests anything.

Reply:
You're right, but I meant that you are not asserting that the values you passed in the constructor are the ones that were set the httpErrorInfo object.